include/swift/Demangling/Demangle.h

```diff
@@ -216,8 +216,21 @@ class Node {
   void dump();
 };
 
+/// Returns the length of the swift mangling prefix of the \p SymbolName.
+///
+/// Returns 0 if \p SymbolName is not a mangled swift (>= swift 4.x) name.
+int getManglingPrefixLength(const char *mangledName);
```

Comment:
Could this be `Optional<StringRef> getManglingPrefix(StringRef)`, where `None` represents no prefix and a `StringRef` is the actual data of the prefix? It seems like all the users are thinking in those terms anyway e.g. the use in `swift-demangler` below could be:

```cpp
auto prefix = swift::Demangle::getManglingPrefix(remangled);
if (!name.startswith(*prefix)) {
    auto namePrefix = swift::Demangle::getManglingPrefix(name);
    remangled =  namePrefix + remangled.substr(prefix->size());
}
```